src/CirclesToken.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity ^0.4.24;
+
+// TODO: Fix external function calls
+// TODO: Ability to update issuance rate
+// TODO: Overflow protection
+import "./CirclesPerson.sol";
+
+contract CirclesToken {
+
+  address person; // Identifier of the token owner
+  uint256 rateUpdatedTimestamp; // Last time the generation rate was updated
+
+  constructor(address _person) public {
+    person = _person;
+    rateUpdatedTimestamp = now;
+  }
+
+  ///////
+  // Optional ERC20 Functions
+  // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
+  ///////
+
+  // TODO: Optional - Figure out address to string conversion
+  //function name() view returns (string name) {
+  //  return "Circles" + person.toString();
+  //}
+
+  // TODO: Optional - Implement?
+  // IDK if there is something short enough that would make sense here
+  //function symbol view() returns (string symbol) {
+  //  return "CIR"
+  //}
+
+  // TODO: Optional - Choose, with issuanceRate
+  function decimals() view returns (uint8 decimals) {
+    return 18;
+  }
+
+  ///////
+  // Required ERC20 functions
+  // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
+  ///////
+
+  //TODO: Choose, with decimals()
+  uint256 constant issuanceRate = 1;
+
+  //TODO: non-continuous payout?
+  function totalSupply() view returns (uint256 totalSupply) {
+    (now - rateUpdatedTimestamp) * issuanceRate
+  }
+
+  uint256 heldElsewhere;
```

Comment:
So "heldElsewhere" isn't the balance of the contract creator, it's the amount _not_ held by the contract creator. It seemed weird to put this in the balances map, because it's the opposite of a balance.

Reply:
Iterating through maps is expensive. Maybe I'm missing your question? What is your proposal?